### PR TITLE
worker(dm): adjust error to warning when bin_log OFF

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close [#3470](https://github.com/pingcap/tiflow/issues/3470)

### Check List 

Tests 

 - Unit test
 - Integration test
 
### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/dm/worker/source_worker.go [hunk 1]
```diff
@@ -175,7 +175,11 @@ func (w *SourceWorker) Start() {
 				}
 			}
 			if err2 := w.updateSourceStatus(w.ctx); err2 != nil {
-				w.l.Error("failed to update source status", zap.Error(err2))
+				if terror.ErrNoMasterStatus.Equal(err2) {
+					w.l.Warn("This source's bin_log is OFF, so it only supports full_mode.", zap.String("sourceID", w.cfg.SourceID), zap.Error(err2))
```
Comment (Ehco1996) [inline dm/dm/worker/source_worker.go:179]: do we need also add this check to pre-check?

Comment (okJiang) [reply dm/dm/worker/source_worker.go:179]: This may happen before task start and after create source. So we can't pre-check it.

Comment (okJiang) [reply dm/dm/worker/source_worker.go:179]: But during the pre-check, we also have `binlogChecker`.

**Review by Ehco1996 — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**